src/app/clusters/icd-management-server/ICDManagementCluster.h

```diff
@@ -17,79 +17,101 @@
-                   uint32_t & icdCounter);
-
-    chip::Protocols::InteractionModel::Status
-    UnregisterClient(chip::app::CommandHandler * commandObj, const chip::app::ConcreteCommandPath & commandPath,
-                     const chip::app::Clusters::IcdManagement::Commands::UnregisterClient::DecodableType & commandData);
-#endif // CHIP_CONFIG_ENABLE_ICD_CIP
+    ICDManagementCluster(EndpointId endpointId, Crypto::SymmetricKeystore & symmetricKeystore, FabricTable & fabricTable,
+                         ICDConfigurationData & icdConfigurationData, OptionalAttributeSet optionalAttributeSet,
+                         BitMask<IcdManagement::OptionalCommands> enabledCommands,
+                         BitMask<IcdManagement::UserActiveModeTriggerBitmap> userActiveModeTriggerBitmap,
+                         CharSpan userActiveModeTriggerInstruction);
 
-private:
-#if CHIP_CONFIG_ENABLE_ICD_CIP
-    /**
-     * @brief Triggers table update events to notify subscribers that an entry was added or removed
-     *        from the ICDMonitoringTable.
-     */
-    void TriggerICDMTableUpdatedEvent();
-#endif // CHIP_CONFIG_ENABLE_ICD_CIP
+    DataModel::ActionReturnStatus ReadAttribute(const DataModel::ReadAttributeRequest & request,
+                                                AttributeValueEncoder & aEncoder) override;
+    CHIP_ERROR Attributes(const ConcreteClusterPath & path, ReadOnlyBufferBuilder<DataModel::AttributeEntry> & builder) override;
 
-    static chip::ICDConfigurationData * mICDConfigurationData;
+    std::optional<DataModel::ActionReturnStatus> InvokeCommand(const DataModel::InvokeRequest & request,
+                                                               TLV::TLVReader & input_arguments, CommandHandler * handler) override;
 
-#if CHIP_CONFIG_ENABLE_ICD_CIP
-    static chip::PersistentStorageDelegate * mStorage;
-    static chip::Crypto::SymmetricKeystore * mSymmetricKeystore;
-#endif // CHIP_CONFIG_ENABLE_ICD_CIP
+    CHIP_ERROR AcceptedCommands(const ConcreteClusterPath & path,
+                                ReadOnlyBufferBuilder<DataModel::AcceptedCommandEntry> & builder) override;
+
+    CHIP_ERROR GeneratedCommands(const ConcreteClusterPath & path, ReadOnlyBufferBuilder<CommandId> & builder) override;
+
+protected:
+#if CHIP_CONFIG_ENABLE_ICD_LIT
+    CHIP_ERROR ReadOperatingMode(AttributeValueEncoder & encoder);
+#endif // CHIP_CONFIG_ENABLE_ICD_LIT
+
+    Crypto::SymmetricKeystore & mSymmetricKeystore;
+    FabricTable & mFabricTable;
+    ICDConfigurationData & mICDConfigurationData;
+    const OptionalAttributeSet mOptionalAttributeSet;
+    const BitMask<IcdManagement::OptionalCommands> mEnabledCommands;
+    const BitMask<IcdManagement::UserActiveModeTriggerBitmap> mUserActiveModeTriggerBitmap;
+    char mUserActiveModeTriggerInstruction[IcdManagement::kUserActiveModeTriggerInstructionMaxLength];
+    const uint8_t mUserActiveModeTriggerInstructionLength;
```

Comment:
Have you checked this ordering of the members for holes?

Reply:
Thanks for pointing out, reordered members.